### PR TITLE
Added auto-reconnect for PostgreSQL database

### PR DESCRIPTION
**Pull request by edewata**
The `PostgreSQLDatabase.connect()` has been added to create
the initial connection, validate the current connection, and
reestablish the connection if it's closed.

### CODE REVIEW

**Comment by frasertweedale** [comment]
Removing myself as reviewer because I have no interest in PostgreSQL DB and no time to test :)

File: base/acme/src/main/java/org/dogtagpki/acme/database/PostgreSQLDatabase.java [hunk 1]
```diff
@@ -82,6 +70,63 @@ public void init() throws Exception {
...
+
+        if (connection == null) { // create the initial connection
+
+            logger.info("Connecting to " + url);
+            connection = DriverManager.getConnection(url, info);
+
+            DatabaseMetaData md = connection.getMetaData();
+            ResultSet rs = null;
+
+            try {
```
Comment (cipherboy) [inline base/acme/src/main/java/org/dogtagpki/acme/database/PostgreSQLDatabase.java:93]: Is there a specific reason we're attempting to get the list of tables? Prove that the connection works? Make sure it is set up (should we be validating the results?).

Comment (edewata) [reply base/acme/src/main/java/org/dogtagpki/acme/database/PostgreSQLDatabase.java:93]: In the original code the connection was created on startup, and this code allowed the admin to verify visually that the database was ready as well. In the new code the connection is created on demand, and now we have auto-recovery too, so this code is not needed anymore. I'll remove it before merging. Thanks!

**Comment by edewata** [comment]
Thanks! Merged.

**Pull request merged**